packages/@aws-cdk/aws-appmesh/test/test.virtual-gateway.ts

```diff
@@ -153,6 +155,174 @@ export = {
       }));
       test.done();
     },
+    'with an http listener with a TLS certificate from ACM'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // WHEN
+      const mesh = new appmesh.Mesh(stack, 'mesh', {
+        meshName: 'test-mesh',
+      });
+
+      const ca = new CfnCertificateAuthority(stack, 'CA', {
+        keyAlgorithm: 'RSA_2048',
+        signingAlgorithm: 'SHA256WITHRSA',
+        subject: {
+          country: 'US',
+          state: 'Washington',
+          locality: 'Seattle',
+          organization: 'App Mesh',
+          organizationalUnit: 'cdk test',
+        },
+        type: 'ROOT',
+      });
+
+      const cert = new CfnCertificate(stack, 'cert', {
+        certificateAuthorityArn: ca.attrArn,
+        certificateSigningRequest: ca.attrCertificateSigningRequest,
+        signingAlgorithm: 'SHA256WITHRSA',
+        validity: {
+          value: 365,
+          type: 'DAYS',
+        },
+      });
+
+      new appmesh.VirtualGateway(stack, 'testGateway', {
+        virtualGatewayName: 'test-gateway',
+        mesh: mesh,
+        listeners: [appmesh.VirtualGatewayListener.httpGatewayListener({
+          port: 8080,
+          tls: {
+            mode: TlsMode.STRICT,
+            certificate: cert,
```

Comment:
As elegant as this would be, looks like I'll be switching to a wrapper object for the ACM cert.